### ISSUE TITLE
Clarify role of language-specific literals

### ISSUE THREAD

**Issue by jamsden** [jamsden/oslc-core-test#59]
When we indicate cardinality of 1 ("exactly one") in a shape, what does this mean from the point of view of language-tagged-strings? For example,

:r1 a oslc:Requirement;  
dcterms:title "my colourful requirement",  
"my colourful requirement"@en-GB,  
"my colorful requirement"@en-US,  
"sorry, don‘t speak french"@fr.

May also be differences from RDF 1.0 to RDF 1.1 in this area.

Where dcterms:title has cardinality 1 on a Requirement. Do we mean "cardinality modulo language tagging"?

**Comment by ndjc** [comment]
I‘m OK with the intent, but the proposed language uses the term ‘plain literal‘, which is not a defined feature of RDF 1.1. Instead, say something like: "For strings and language-tagged strings, single-valued means there is at most one value for any given language tag, and at most one untagged value".

**Comment by DavidJHoney** [comment]
Proposal:

1) Treat OSLC cardinality as applying solely to plain literals.  
2) Allow OSLC data providers to include additional literals with language tags.  
3) Recommend that OSLC clients that don‘t understand language tags should only process plain literals.

**Comment by ndjc** [comment]
I believe the above interpretation of OSLC 2 is too strict. It is true that rdf:langString is not one of the documented oslc:valueType property values, so you cannot constrain a string property to be a tagged string.

While it is true that xsd:string and rdf:langString are completely different types (one of the absurdities of RDF), there is plenty of existing string data that uses both tagged and untagged strings for the same property - including data published by tools that have OSLC clients and servers. As a result, I believe we would only be admitting existing practice by explicitly describing this use of multiple languages for a property with an occurrence of zero-or-one, or exactly-one.

**Comment by ndjc** [comment]
I do not know of tools that use XMLLiteral to embed language attributes in the way you describe - the result would be unnecessarily hard to parse. XMLLiteral is commonly used in OSLC properties to allow rich text markup for display, not (in my experience) for language tags. I am not aware of any tools that **currently** publish OSLC-defined string properties that use both tagged and untagged strings; the examples I mentioned in my previous comment were in vocabularies. However, we have had discussions about having tagged strings for titles, names, descriptions, and comments in some tools, and the fact that OSLC does not correctly support this was not considered - had it been, we would have considered it a bug with OSLC!

In retrospect, OSLC should have defined an oslc:valueType of ‘any string‘ that would include xsd:string and rdf:langString. If we stick to the strict interpretation of xsd:string, then I would argue for adding this new value in 3.0 - but it might be difficult to go back and redefine many existing OSLC shapes to use this new value in place of xsd:string.

**Comment by ndjc** [comment]
For the purposes of the ShapeChecker tool, I have interpreted ‘exactly-one‘ for string literals as being modulo language tags - the checker will complain if there are two or more untagged literals, or two or more of the same exact language tag. I believe this interpretation reflects the underlying intent of the cardinality constraint, even if it does not match it mathematically.

**Comment by DavidJHoney** [comment]
I agree with Nick. The obvious use case for cardinality constraints is that a client is looking for a maximum of one literal for some client locale. If the client is not interested in language specific literals, they will want one xsd:string value. If the client is interested in language specific literals, they will want to try to get one literal for their client locale, and if none exists, use a maximum of one xsd:string value. This is such a common usage pattern generally, I think OSLC should support it.
